src/operator/tensor/dot-inl.h

```diff
@@ -45,13 +45,20 @@ namespace op {
 struct DotParam : public dmlc::Parameter<DotParam> {
   bool transpose_a;
   bool transpose_b;
+  dmlc::optional<int> forward_stype_hint;
   DMLC_DECLARE_PARAMETER(DotParam) {
     DMLC_DECLARE_FIELD(transpose_a)
       .describe("If true then transpose the first input before dot.")
       .set_default(false);
     DMLC_DECLARE_FIELD(transpose_b)
       .describe("If true then transpose the second input before dot.")
       .set_default(false);
+    DMLC_DECLARE_FIELD(forward_stype_hint)
+      .describe("Desired storage type of the forward output.")
```

Comment:
Need to be more detailed than this. Explicitly state that if no such combination is implemented, dense op is used.